presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaRecordSet.java

```diff
@@ -305,30 +353,28 @@ private void checkFieldType(int field, Class<?> expected)
         @Override
         public void close()
         {
+            log.debug("Close consumer if exist: %d, partition %d", Thread.currentThread().getId(), split.getPartitionId());
+            if (this.consumer != null) {
+                this.consumer.close();
+            }
         }
 
         private void openFetchRequest()
         {
             try {
                 if (messageAndOffsetIterator == null) {
-                    log.debug("Fetching %d bytes from offset %d (%d - %d). %d messages read so far", KAFKA_READ_BUFFER_SIZE, cursorOffset, split.getStart(), split.getEnd(), totalMessages);
-                    FetchRequest req = new FetchRequestBuilder()
-                            .clientId("presto-worker-" + Thread.currentThread().getName())
-                            .addFetch(split.getTopicName(), split.getPartitionId(), cursorOffset, KAFKA_READ_BUFFER_SIZE)
-                            .build();
-
-                    // TODO - this should look at the actual node this is running on and prefer
-                    // that copy if running locally. - look into NodeInfo
-                    SimpleConsumer consumer = consumerManager.getConsumer(split.getLeader());
-
-                    FetchResponse fetchResponse = consumer.fetch(req);
-                    if (fetchResponse.hasError()) {
-                        short errorCode = fetchResponse.errorCode(split.getTopicName(), split.getPartitionId());
-                        log.warn("Fetch response has error: %d", errorCode);
-                        throw new RuntimeException("could not fetch data from Kafka, error code is '" + errorCode + "'");
+                    long threadId = Thread.currentThread().getId();
+
+                    if (this.consumer == null) {
+                        this.consumer = consumerManager.createConsumer(threadId, split.getLeader());
                     }
 
-                    messageAndOffsetIterator = fetchResponse.messageSet(split.getTopicName(), split.getPartitionId()).iterator();
+                    TopicPartition tp = new TopicPartition(split.getTopicName(), split.getPartitionId());
+                    this.consumer.assign(ImmutableList.of(tp));
+                    this.consumer.seek(tp, cursorOffset);
+                    ConsumerRecords<ByteBuffer, ByteBuffer> records = this.consumer.poll(3000);
+                    this.consumer.commitSync();
```

Comment:
why need commitSync?

Reply:
Good point. It's not required. I copied over from a consumer example.